src/basic/time-util.h

```diff
@@ -112,10 +113,15 @@ static inline bool triple_timestamp_is_set(const triple_timestamp *ts) {
 usec_t triple_timestamp_by_clock(triple_timestamp *ts, clockid_t clock);
 
 usec_t timespec_load(const struct timespec *ts) _pure_;
+_pure_ static inline usec_t statx_timestamp_load(const struct statx_timestamp *ts) {
+        return timespec_load(&(const struct timespec) { .tv_sec = ts->tv_sec, .tv_nsec = ts->tv_nsec });
+}
 nsec_t timespec_load_nsec(const struct timespec *ts) _pure_;
+_pure_ static inline nsec_t statx_timestamp_load_nsec(const struct statx_timestamp *ts) {
```

Comment:
i'd drop the `_pure_`, the compiler should always be able to figure that out on its own here. 

the `_pure_` stuf makes sense for functions whose implementation is hidden away in .c files, where the compiler cannot look into (because it becomes available only during link time). But for static inline stuff defined in headers this is unnecessary, the compiler can always make the determinations on its own, because it has the full sources around.